oauthproxy_test.go

```diff
@@ -2938,12 +2938,20 @@ func TestProxyAllowedGroups(t *testing.T) {
 		name               string
 		allowedGroups      []string
 		groups             []string
+		querystring        string
 		expectUnauthorized bool
 	}{
-		{"NoAllowedGroups", []string{}, []string{}, false},
-		{"NoAllowedGroupsUserHasGroups", []string{}, []string{"a", "b"}, false},
-		{"UserInAllowedGroup", []string{"a"}, []string{"a", "b"}, false},
-		{"UserNotInAllowedGroup", []string{"a"}, []string{"c"}, true},
+		{"NoAllowedGroups", []string{}, []string{}, "", false},
+		{"NoAllowedGroupsUserHasGroups", []string{}, []string{"a", "b"}, "", false},
+		{"UserInAllowedGroup", []string{"a"}, []string{"a", "b"}, "", false},
+		{"UserNotInAllowedGroup", []string{"a"}, []string{"c"}, "", true},
+		{"UserInQuerystringGroup", []string{"a", "b"}, []string{"a", "c"}, "?allowed_groups=a", false},
+		{"UserInMultiParamQuerystringGroup", []string{"a", "b"}, []string{"b"}, "?allowed_groups=a&allowed_groups=b,d", false},
+		{"UserInOnlyQuerystringGroup", []string{}, []string{"a", "c"}, "?allowed_groups=a,b", false},
+		{"UserInDelimitedQuerystringGroup", []string{"a", "b", "c"}, []string{"c"}, "?allowed_groups=a,c", false},
+		{"UserNotInQuerystringGroup", []string{}, []string{"c"}, "?allowed_groups=a,b", true},
+		{"UserInConfigGroupNotInQuerystringGroup", []string{"a", "b", "c"}, []string{"c"}, "?allowed_groups=a,b", true},
+		{"UserInQuerystringGroupNotInConfigGroup", []string{"a", "b"}, []string{"c"}, "?allowed_groups=b,c", true},
```

Comment:
This gets really hard to read and mentally parse. Can you please explicitly name the keys?

```suggestion
		{
				name: "NoAllowedGroups",
				allowedGroups: []string{},
				groups: []string{},
				querystring: "",
				expectedUnauthorized: false,
		},
		...
```